src/main/java/com/google/sps/servlets/JoinSessionServlet.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Attendee;
+import java.util.Date;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.DatastoreClientInterface;
+import com.google.sps.data.DatastoreClient;
+import com.google.sps.data.AttendeeInterface;
+import com.google.sps.data.Attendee;
+
+/** Servlet that returns a list of attendees in a session.  */
```

Comment:
This isn't an accurate comment about what this class does. Based on your design doc:
"their info is stored in datastore and the user is redirected to the session webpage where they are connected to the session w/ noVNC (by getting information about ipOfVM from datastore)"

Im also not sure why there is a doGet function. Join session doesnt need to get anything

Reply:
I had the doGet get the list of attendees to check if the username is unique to the session. But I'll get the list from session.js instead

Reply:
yeah, theres no need for a doGet(). You can get the session from the server_client that Chris wrote